Objects/regions.c

```diff
@@ -80,10 +92,20 @@ struct PyRegionObject {
 };
 
 struct regionmetadata {
-    int lrc;  // Integer field for "local reference count"
-    int osc;  // Integer field for "open subregion count"
+    // The number of references coming in from the local region.
+    PY_UINT32_T lrc;
+    // The number of open subregions.
+    PY_UINT32_T osc;
+    // The number of objects inside this region.
+    PY_UINT32_T rc;
     int is_open;
-    regionmetadata* parent;
+    // This field might either point to the parent region or another region that
+    // this one was merged into. The `Py_METADATA_MERGE_TAG` tag is used to indicate
+    // the this points to a merged region.
+    Py_region_ptr_with_tags_t parent;
+    // A weak reference to the bridge object. The bridge object has increased the
```

Comment:
We're cheating :D

I assume that we would need to actually allocate an object for Python's weak reference. Since the GC can't see this type I feel like it's not worth it